### ISSUE TITLE
ytdl_hook-exclude works in command line but doesn't work in config file

### ISSUE THREAD

**Issue by nzm001** [mpv-player/mpv#11046]
### Important Information

Provide following Information:
- mpv version: release 0.35.0
- Platform and Version: win10 and ubuntu20.04
- Source of the mpv binary: https://sourceforge.net/projects/mpv-player-windows/files, https://non-gnu.uvt.nl/debian

### Reproduction steps

install python, install yt-dlp (pip)

config file in portable_config or ~/.config/mpv/

```
script-opts-append=ytdl_hook-try_ytdl_first=yes
script-opts-append=ytdl_hook-exclude=".*bilibili.*" # does not work
script-opts-append=ytdl_hook-all_formats=yes # works, proving config file works
script-opts-append=ytdl_hook-force_all_formats=no
script-opts-append=ytdl_hook-use_manifests=yes
script-opts-append=ytdl_hook-ytdl_path=yt-dlp
```

shell

```sh
# windows
> .\mpv.com --no-config https://www.bilibili.com/video/BV1ZG4y1J7yd # Expected
playing
> .\mpv.com --no-config https://www.bilibili.com/video/BV1ZG4y1J7yd --script-opts-append=ytdl_hook-exclude=".*bilibili.*" # Expected
Failed to recognize file format.

Exiting... (Errors when loading file)
> .\mpv.com https://www.bilibili.com/video/BV1ZG4y1J7yd # Not Expected
playing

# linux
$ mpv --no-config https://www.bilibili.com/video/BV1ZG4y1J7yd # Expected
playing
$ mpv --no-config https://www.bilibili.com/video/BV1ZG4y1J7yd --script-opts-append=ytdl_hook-exclude=".*bilibili.*" # Expected
Failed to recognize file format.

Exiting... (Errors when loading file)
$ mpv.com https://www.bilibili.com/video/BV1ZG4y1J7yd # Not Expected
playing
```

**Comment by avih** [comment]
> `script-opts-append=ytdl_hook-exclude=".*bilibili.*" # does not work`

The problem here is that you think mpv config parsing is the same as shell parsing as far as quoting goes, but it's not.

Mpv parsing does not support mixed quoting, e.g. these are the same in shell:
```sh
foo --bar="baz.*"
foo "--bar=baz.*"
```

But in mpv config, the variable is `script-opts-append`, and the value should be `ytdl_hook-exclude=.*bilibili.*`, however, you quoted only part of the value (mixed quoting), and it ends up with the quotes becoming part of the value.

You need to quote the wole value, or not use quotes at all, because mpv parsing does not expand `*` glob, so this specific line doesn't need the quotes:
```
script-opts-append="ytdl_hook-exclude=.*bilibili.*"
```

**Comment by avih** [comment]
Actually, it's a bit more complex than that.

`script-opts` is a key-value list - https://mpv.io/manual/master/#key-value-list-options

And `-append` is for a single vaue (in your case `ytdl_hook-exclude=.*bilibili.*`) where escapes are not interpreted, while `-add` is for multiple values where the separator can be escaped.

So somehow, for add, you can use quotes for each value, e.g.
```
script-opts-add=foo="x",bar="y"
```
Where `script-opts/foo` is `x`, and `script-opts/bar` is `y`.

But for append:
```
script-opts-append=foo="x",bar="y"
```
ends up with `script-opts/foo` being `"x",bar="y"` (quotes are part of the value).

So basically, `-append` needs either the whole `sub=value` quoted, or not quoted at all.
